opentelemetry-sdk/src/opentelemetry/sdk/_metrics/metric_reader_storage.py

```diff
@@ -44,27 +50,58 @@ def _get_or_init_view_instrument_match(
         # Optimistically get the relevant views for the given instrument. Once set for a given
         # instrument, the mapping will never change
 
-        if instrument in self._view_instrument_match:
-            return self._view_instrument_match[instrument]
+        if instrument in self._instrument_view_instrument_matches:
+            return self._instrument_view_instrument_matches[instrument]
 
         with self._lock:
             # double check if it was set before we held the lock
-            if instrument in self._view_instrument_match:
-                return self._view_instrument_match[instrument]
+            if instrument in self._instrument_view_instrument_matches:
+                return self._instrument_view_instrument_matches[instrument]
 
             # not present, hold the lock and add a new mapping
             view_instrument_matches = []
             for view in self._sdk_config.views:
                 # pylint: disable=protected-access
                 if view._match(instrument):
-                    view_instrument_matches.append(
-                        _ViewInstrumentMatch(
-                            view=view,
-                            instrument=instrument,
-                            sdk_config=self._sdk_config,
-                        )
+
+                    new_view_instrument_match = _ViewInstrumentMatch(
+                        view=view,
+                        instrument=instrument,
+                        sdk_config=self._sdk_config,
                     )
 
+                    if isinstance(instrument, Asynchronous) and isinstance(
```

Comment:
I might be missing something but why is there a check for this specific instance of view configuration and instrument type being Async?

Reply:
It is a particular error mentioned [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view):

> (e.g. the View sets an asynchronous instrument to use the Explicit bucket histogram aggregation)

Reply:
Isn't that just an example? Why specifically check this?

Reply:
Well, it may be an example, but we have to check that? That does not mean that is all we have to check, of course.

Reply:
Resolving this conversation as the same topic is being discussed below.